pkg/operatorrules/registry_test.go

```diff
@@ -176,5 +178,37 @@ var _ = Describe("OperatorRules", func() {
 			Expect(registeredAlerts[0].Expr.String()).To(Equal("sum(rate(http_requests_total[1m])) > 100"))
 			Expect(registeredAlerts[1].Expr.String()).To(Equal("sum(rate(http_requests_total[1m])) > 200"))
 		})
+
+		It("should create 2 alerts when registered with the same name but different expressions", func() {
+			alerts := []promv1.Rule{
+				{
+					Alert: "ExampleAlert1",
+					Expr:  intstr.FromString("sum(rate(http_requests_total[1m])) > 100"),
+				},
+			}
+
+			err := or.RegisterAlerts(alerts)
+			Expect(err).To(BeNil())
+
+			alerts = []promv1.Rule{
+				{
+					Alert: "ExampleAlert1",
+					Expr:  intstr.FromString("sum(rate(http_requests_total[1m])) > 200"),
+				},
+			}
+
+			err = or.RegisterAlerts(alerts)
+			Expect(err).To(BeNil())
+
+			Consistently(func(g Gomega) {
+				registeredAlerts := or.ListAlerts()
+				g.Expect(registeredAlerts).To(HaveLen(2))
+				g.Expect(registeredAlerts[0].Expr.String()).To(Equal("sum(rate(http_requests_total[1m])) > 100"))
+				g.Expect(registeredAlerts[1].Expr.String()).To(Equal("sum(rate(http_requests_total[1m])) > 200"))
+			}).
+				WithTimeout(20 * time.Millisecond).
+				WithPolling(1 * time.Millisecond).
+				Should(Succeed())
```

Comment:
Constantly is used for async operations. A simple golang loop can do the same, if you just want to repeat.